source/fe/fe_data.cc

```diff
@@ -30,6 +30,8 @@ FiniteElementData<dim>::FiniteElementData(
   , dofs_per_line(dofs_per_object[1])
   , dofs_per_quad(dim > 1 ? dofs_per_object[2] : 0)
   , dofs_per_hex(dim > 2 ? dofs_per_object[3] : 0)
+  , non_local_dofs_per_cell(
+      dofs_per_object.size() == dim + 2 ? dofs_per_object[dim + 1] : 0)
```

Comment:
this convention of possibly tacking on an element to the array needs to be documented

Reply:
we have documented this, please have a look.